app/services/foreman/template_snapshot_service.rb

```diff
@@ -105,6 +105,7 @@ def host4dhcp
         name: 'snapshot-ipv4-dhcp-el7',
         subnet: FactoryBot.build(:subnet_ipv4_dhcp_for_snapshots),
         interfaces: [ipv4_interface])
+      host.stubs(:managed_interfaces).returns(host.interfaces)
```

Comment:
I'd prefer to build a host with the right interfaces. I think it means `ipv4_interface` needs to be marked as `managed`.

Reply:
`managed_interfaces` is based on DB scopes, it does not work correctly for non-saved hosts:
``` ruby
      host = FactoryBot.build(:managed_host,
        name: 'snapshot-ipv6-dhcp-el7',
        interfaces: [FactoryBot.build(:managed_interface, managed: true)])
      host.managed_interfaces
      # => []

      host.save!
      host.managed_interfaces
      # => [<Nic::Managed ... >]
```